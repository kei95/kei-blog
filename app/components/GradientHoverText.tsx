import React from "react";

interface GradientHoverTextProps {
  title: string;
  className?: React.DetailedHTMLProps<
    React.HTMLAttributes<HTMLHeadingElement>,
    HTMLHeadingElement
  >["className"];
}

export default function GradientHoverText({
  title,
  className,
}: GradientHoverTextProps) {
  const styleClassName =
    "bg-clip-text text-transparent bg-white hover:bg-gradient-to-r from-[#edca85] via-[#d55594] to-[#5594d5] " +
    className;
  return <h3 className={styleClassName}>{title}</h3>;
}
