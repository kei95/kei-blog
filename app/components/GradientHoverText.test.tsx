import { render, screen } from "@testing-library/react";

import GradientHoverText from "./GradientHoverText";

describe("<GradientHoverText />", () => {
  test("render the component - it should match the snapshot", () => {
    const mockedText = "Lorem Ipsum";

    render(<GradientHoverText title={mockedText} className="px-4" />);
    expect(screen.getByText(mockedText)).toMatchSnapshot();
  });
});
