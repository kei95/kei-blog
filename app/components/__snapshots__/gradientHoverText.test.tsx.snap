// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GradientHoverText /> render the component - it should match the snapshot 1`] = `
<h3
  class="bg-clip-text text-transparent bg-white hover:bg-gradient-to-r from-[#edca85] via-[#d55594] to-[#5594d5] px-4"
>
  Lorem Ipsum
</h3>
`;
